BaseStation/BaseServer/BaseServer.cs

```diff
@@ -1,11 +1,147 @@
-﻿using System.Net;
+﻿using Scarlet.Communications;
+using Scarlet.Utilities;
+using SharpDX.XInput;
+using System;
+using System.Net;
+using System.Linq;
+using System.Threading;
+using System.Collections.Generic;
 
-namespace HuskyRobotics.BaseStation.Server {
+namespace HuskyRobotics.BaseStation.Server
+{
     /// <summary>
     /// Contains functionality to communicate with the client/rover.
     /// Acts as an abstraction over Scarlet communications API.
     /// </summary>
-    public static class BaseServer {
+    public static class BaseServer
+    {
 
+        private static bool shutdown = false;
+        private static Controller gamepad;
+        private static int LeftThumbDeadzone = 7849;
+        private static int RightThumbDeadzone = 8689;
+        private static int TriggerThreshold = 30;
+
+        public static void Start()
+        {
+            Scarlet.Communications.Server.Start(1025, 1026);
+            Scarlet.Communications.Server.ClientConnectionChange += memes;
+            Parse.SetParseHandler(0xC0, gpsHandler);
+            Parse.SetParseHandler(0xC1, magnetomerHandler);
+            gamepad = GetGamepad();
+        }
+
+        private static void memes(object sender, EventArgs e)
+        {
+            Console.WriteLine("Clients Changed");
+            Console.WriteLine(Scarlet.Communications.Server.GetClients());
+        }
+
+        private static Controller GetGamepad()
+        {
+            Controller c = new Controller(UserIndex.One);
+            return c;
+        }
+
+        public static void EventLoop()
+        {
+            while (!shutdown)
+            {
+                if (!gamepad.IsConnected)
+                {
+                    Console.WriteLine("Gamepad not connected");
+                    Thread.Sleep(100);
+                    continue;
+                }
+                State state = gamepad.GetState();
+                byte rightTrigger = state.Gamepad.RightTrigger;
+                byte leftTrigger = state.Gamepad.LeftTrigger;
+                short leftThumbX = state.Gamepad.LeftThumbX;
+
+                Console.WriteLine(leftThumbX);
+
+                if (leftThumbX == -32768)
+                {
+                    leftThumbX++;
+                }
+
+                if (rightTrigger < TriggerThreshold) { rightTrigger = 0; }
+                if (leftTrigger < TriggerThreshold) { leftTrigger = 0; }
+                if (Math.Abs(leftThumbX) < LeftThumbDeadzone) { leftThumbX = 0; }
+
+                float speed = rightTrigger - leftTrigger;
```

Comment:
Instead of assigning these variables here, take the rhs and put it in the LinearMap. That avoids assigning to the same variable twice which can be more difficult to reason about